### ISSUE TITLE
random doesn't work in OpenFool 0.0.9 Qt 5.5.1 win32

### ISSUE THREAD

**Issue by artenax** [hyst329/OpenFool#50]
Приветствую.
У меня вопрос по Qt5 версии (0.0.9), я знаю, вы ее уже не разрабатываете, но может подскажете. У вас на [странице релизов](https://github.com/trolley813/OpenFool/releases/tag/v0.0.9) есть сборки для Android и Win7+ (Qt 5.8). Поскольку я пользуюсь Ubuntu 16.04 i386 (Qt 5.5.1), я скомпилировал для себя, приложение запускается, random работает. У меня нет Win7, но в вашей сборке [OpenFool-0.0.9-windows.7z](https://github.com/trolley813/OpenFool/releases/download/v0.0.9/OpenFool-0.0.9-windows.7z) (Qt 5.8, Win7+), запущенной в Wine, random тоже работает. Однако я хотел сделать сборку для пользователей Windows XP (у меня есть в dualboot). Последний Qt, который ее поддерживает, это Qt 5.6. Но реально запустить оказалось только QtCreator 3.5.1 с Qt 5.5.1 на XP. Игра скомпилировалась, на XP запускается, но я заметил такую проблему. При каждом запуске игры одни и те же карты, т.е. не работает random. Я проверил в виртуальной XP - те же карты, проверил в Wine - те же карты. Так что здесь есть:
1. либо какой-то баг, который проявляется в win32 + Qt 5.5.1 (потому что в линуксовом Qt 5.5.1 всё ok)
2. может быть я не установил какой-то SDK перед сборкой

Я нашел упоминание random в файле [carddeck.cpp](https://github.com/trolley813/OpenFool/blob/11c3a4f20361ea6ecacbb17903d24149c88047a9/carddeck.cpp). Это `#include <random>`, CryptAcquireContextW и CryptGenRandom. Это майкрософтовские функции, которые поддерживаются в XP и Wine, но почему-то не срабатывают у меня. Не срабатывают именно в сборке [win32 + Qt 5.5.1](https://github.com/artenax/openfool/releases). Причем, и в Wine.
Не подскажете в чем причина и как исправить, чтобы random работал?

Я знаю, конечно, Qt5 версия недоделана. Там, например, нельзя менять количество игроков. В моей сборке почему-то не применяется локализация (это мелочи). Но игра мне понравилась. И хотя я пользуюсь в основном Linux, где всё ok, я бы хотел также чтобы XP версия (Qt5.5-5.6) нормально работала. Может быть икспишникам понравится.

**Comment by trolley813** [comment]
Здравствуйте! А каким компилятором вы пользуетесь?

**Comment by artenax** [comment]
Приветствую.
На XP я пользуюсь компилятором из состава [Qt 5.5.1](https://download.qt.io/new_archive/qt/5.5/5.5.1/qt-opensource-windows-x86-mingw492-5.5.1.exe) bundle:
mingw 4.9.2 (i686-posix-dwarf-rev1, Built by MinGW-W64 project), (C) 2014

В Ubuntu 16.04 (где random работает):
gcc 5.4.0 (20160609, Ubuntu), (C) 2015

**Comment by artenax** [comment]
Попровал подменить на версию [5.4.0](https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win32/Personal%20Builds/mingw-builds/5.4.0/threads-posix/dwarf/), но не получилось. Собирается, но не запускается. Виндовый Qt5 привязан к версии mingw, с которой он поставляется.

**Comment by trolley813** [comment]
Да, вероятно, версия 4.9.2 очень старая, там еще толком не поддерживается стандарт С++11. Возможно, дело связано с этим
